python_modules/dagster/dagster/_cli/utils.py

```diff
@@ -1,35 +1,68 @@
+import logging
 import os
 import tempfile
 from contextlib import contextmanager
-from typing import Callable, Iterator, Optional
-
-import click
+from typing import Iterator, Optional
 
 from dagster._core.instance import DagsterInstance, InstanceRef
 from dagster._core.instance.config import is_dagster_home_set
+from dagster._core.secrets.env_file import get_env_var_dict
 
 
 @contextmanager
-def get_instance_for_service(
-    service_name: str,
+def get_instance_for_cli(
+    cli_command: Optional[str] = "this command",
     instance_ref: Optional[InstanceRef] = None,
-    logger_fn: Callable[[str], None] = click.echo,
+    fallback_to_tempdir: bool = False,
```

Comment:
Can we push this fall back logic to the single callsite that uses it rather than adding another parameter here? I think adding these one-off booleans is q code smell and we should avoid it if possible

Reply:
I count 7 callsites (basically everywhere that was using this old method that I have now repurposed - for CLI entrypoints where we are comfortable creating an ephemeral instance if no DAGSTER_HOME is set) - I could add a separate method instead? get_instance_for_cli (default) / get_possibly_ephemeral_instance_for_cli?